File: python/bob/bob.py
def is_yelled(string: str) -> bool:
    return string.isupper()


def is_question(string: str) -> bool:
    return string.endswith("?")


def is_yelled_question(string: str) -> bool:
    return is_yelled(string) and is_question(string)


def hey(phrase: str) -> str:
    phrase = phrase.strip()
    if is_yelled_question(phrase):
        return "Calm down, I know what I'm doing!"
    elif is_yelled(phrase):
        return "Whoa, chill out!"
    elif is_question(phrase):
        return "Sure."
    elif not phrase:
        return "Fine. Be that way!"
    else:
        return "Whatever."

File: go/bob/bob.go
// Package bob contains all the functions to simulate Bob the teenager
package bob

import "strings"

func sanitize(str string) string {
	return strings.TrimSpace(str)
}

func isUpper(str string) bool {
	return str == strings.ToUpper(str) && str != strings.ToLower(str)
}

func isBlank(str string) bool {
	return str == ""
}

func isQuestion(str string) bool {
	return strings.HasSuffix(str, "?")
}

func isYelledSentence(str string) bool {
	return isUpper(str) && !isQuestion(str)
}

func isYelledQuestion(str string) bool {
	return isUpper(str) && isQuestion(str)
}

// Hey simulates Bob, the lackadaisical teenager
func Hey(remark string) string {
	remark = sanitize(remark)
	switch {
	case isYelledSentence(remark):
		return "Whoa, chill out!"
	case isYelledQuestion(remark):
		return "Calm down, I know what I'm doing!"
	case isQuestion(remark):
		return "Sure."
	case isBlank(remark):
		return "Fine. Be that way!"
	}
	return "Whatever."
}
